### PR TITLE
Display nicer names for github-downloaded databases

### PR DESCRIPTION
**Pull request by aeisenberg**
This will now name databases downloaded from github based on their nwo.

Also, this adds a new button to suggest downloading from github in an
empty databases view.

Replace this with a description of the changes your pull request makes.

## Checklist

- [n/a] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## [UNRELEASED]
 
 - Re-enable publishing to open-vsx. [#1285](https://github.com/github/vscode-codeql/pull/1285)
+- Databases downloaded from GitHub have a more user friendly name. These databases will now be named `{org}/{repo}` instead of `codeql-db`. [#1294](https://github.com/github/vscode-codeql/pull/1294)
```
Comment (shati-patel) [inline extensions/ql-vscode/CHANGELOG.md:6]: No need for a changelog entry - downloading from GitHub is still a hidden feature (due to the MRVA API being restricted)

File: extensions/ql-vscode/package.json [hunk 2]
```diff
@@ -1089,7 +1089,7 @@
       },
       {
         "view": "codeQLDatabases",
-        "contents": "Add a CodeQL database:\n[From a folder](command:codeQLDatabases.chooseDatabaseFolder)\n[From an archive](command:codeQLDatabases.chooseDatabaseArchive)\n[From a URL (as a zip file)](command:codeQLDatabases.chooseDatabaseInternet)\n[From LGTM](command:codeQLDatabases.chooseDatabaseLgtm)"
+        "contents": "Add a CodeQL database:\n[From GitHub](command:codeQLDatabases.chooseDatabaseGithub)\n[From a folder](command:codeQLDatabases.chooseDatabaseFolder)\n[From an archive](command:codeQLDatabases.chooseDatabaseArchive)\n[From a URL (as a zip file)](command:codeQLDatabases.chooseDatabaseInternet)\n[From LGTM](command:codeQLDatabases.chooseDatabaseLgtm)"
```
Comment (shati-patel) [inline extensions/ql-vscode/package.json:1092]: As above! Unless there's a way to make this button canary-only, we should hide it until the feature is more widely available.

File: extensions/ql-vscode/src/databases.ts [hunk 3]
```diff
@@ -569,14 +571,15 @@ export class DatabaseManager extends DisposableObject {
...
+    displayName?: string
   ): Promise<DatabaseItem> {
     const contents = await resolveDatabaseContents(uri);
     // Ignore the source archive for QLTest databases by default.
     const isQLTestDatabase = path.extname(uri.fsPath) === '.testproj';
     const fullOptions: FullDatabaseOptions = {
       ignoreSourceArchive: isQLTestDatabase,
-      // displayName is only set if a user explicitly renames a database
-      displayName: undefined,
+      // If a diaplayName is not passed in, the basename of folder containing the database is used.
```
Comment (shati-patel) [inline extensions/ql-vscode/src/databases.ts:581]: Minor :)

```suggestion
      // If a displayName is not passed in, the basename of the folder containing the database is used.
```

**Review by shati-patel — COMMENTED** [review]
Much nicer, thanks 😄 A few minor changes, since this is still an internal-only feature.

File: extensions/ql-vscode/CHANGELOG.md (hunk 1 above)
Comment (aeisenberg) [reply extensions/ql-vscode/CHANGELOG.md:6]: Right!

File: extensions/ql-vscode/package.json (hunk 2 above)
Comment (aeisenberg) [reply extensions/ql-vscode/package.json:1092]: Hmmm...no way to do that. I'll remove. I forgot it is still a canary-only feature.

**Review by shati-patel — APPROVED** [review]
:shipit:

**Pull request merged**